File: Java/0001-0050/1. Two Sum.java
// https://leetcode.com/problems/two-sum/description/

class Solution {
    public int[] twoSum(int[] nums, int target) {
        //store all value and index in HashMap
        HashMap<Integer, Integer> storage = new HashMap<>();

        int [] result = new int[2];
        //iterate the array to find the result 
        for(int index=0; index<nums.length; index++){
          if(storage.containsKey( target - nums[index])){
              result[0] = index;
              result[1] = storage.get(target - nums[index]);
          }
          //add the value to hashMap
          storage.put(nums[index],index);
        }

        return result;
    }
}

File: Python/0001-0050/1. Two Sum.py
# https://leetcode.com/problems/two-sum/description/

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        # store all value and index in Dictionary
        storage = {}

        result = [-1,-1]
        # iterate the array to find the result 
        for index in range(0, len(nums)):
            if (target - nums[index]) in storage:
                result[0] = index
                result[1] = storage[target - nums[index]]
            
            # add the value to Dictionary
            storage[nums[index]] = index

        return result;
      
